### DISCUSSION TITLE
Other Segmentation Tools

### DISCUSSION THREAD

**Discussion by e-hossam96** [aub-mind/arabert#202] [Q&A] [answered]
Hello all,

Thanks for the good effort. Is it OK to use other Disambiguation/Segmentation tools like Camel-Tools to preprocess input text before passing it to the AraBERTv2 tokenizer?

Thanks

**Comment by WissamAntoun** [comment, marked as answer]
Hey Hossam, I haven't tried replacing the pre-processor yet. In any case, I'd recommend against it, since even if the Camel Tool produces the same prefixes or suffixes if the tokenization of words let's say is more accurate, it will be different than what was seen during pretraining, and hence the performance might be slightly worst and not optimal.

**Reply by e-hossam96** [reply]
Very insightful. Thanks for your prompt reply.
